File: semestr-8/Programowanie obiektowe w Java/alexander/POJ/2023s/IZ08IO2/BridgeProject/src/ShapeRectangle.java

import java.awt.Color;
import java.awt.Graphics;

public class ShapeRectangle extends ShapeBase
{
    private int ax,ay,bx,by;
    public ShapeRectangle(String text)
    {
        parse(text, (v)->ax=v, (v)->ay=v, (v)->bx=v, (v)->by=v);
    }
    
    @Override public void Draw(Graphics g)
    {
        g.setColor(Color.blue);
        g.drawRect(ax,ay,bx,by);
    }
    
    @Override public String toXML()
    {
        return String.format("<rectangle ax=\"%d\" ay=\"%d\" bx=\"%d\" by=\"%d\" />",ax,ay,bx,by);
    }

    @Override public String toString()
    {
        return String.format("Rectangle(ax:%d,ay:%d,bx:%d,by:%d)",ax,ay,bx,by);
    }
}

File: semestr-8/Jezyk C#/alexander/PCS/2023s/ID06TC1/BridgePattern/BridgePattern/ShapeRectangle.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BridgePattern
{
    class ShapeRectangle : ShapeBase
    {
        private int ax, ay, bx, by;
        public ShapeRectangle(string parameters)
        {
            Parse(parameters, (v) => ax = v, (v) => ay = v, (v) => bx = v, (v) => by = v);
        }
        public override void Draw(Graphics g)
        {
            g.DrawRectangle(Pens.Blue, ax, ay, bx - ax, by - ay);
        }

        public override string ToXml()
        {
            return $"<rectangle ax=\"{ax}\" ay=\"{ay}\" bx=\"{bx}\" by=\"{by}\" />";
        }
        public override string ToString()
        {
            return $"Rectangle(cx:{ax},cy:{ay},bx:{bx},by:{by})";
        }
    }
}
